crates/types/src/vote.rs

```diff
@@ -28,19 +28,34 @@ use crate::{
 };
 
 /// A simple vote that has a signer and commitment to the data voted on.
-pub trait Vote<TYPES: NodeType>: HasViewNumber<TYPES> {
+pub trait Vote<TYPES: NodeType>: HasViewNumber<TYPES> + Committable {
     /// Type of data commitment this vote uses.
-    type Commitment: Voteable;
+    type Data: Voteable;
 
     /// Get the signature of the vote sender
     fn signature(&self) -> <TYPES::SignatureKey as SignatureKey>::PureAssembledSignatureType;
     /// Gets the data which was voted on by this vote
-    fn date(&self) -> &Self::Commitment;
-    /// Gets the Data commitment of the vote
-    fn date_commitment(&self) -> Commitment<Self::Commitment>;
+    fn data(&self) -> &Self::Data;
+    /// Gets the (Data + view number) commitment of the vote
+    fn vote_commitment(&self) -> Commitment<Self>;
```

Comment:
Same as other comment if `Vote` must be committable then this is redundant right?